src/push/apns.rs

```diff
@@ -45,10 +46,22 @@ pub fn send_push(
     collapse_id: Option<CollapseId>,
     ttl: u32,
 ) -> SendFuture<(), SendPushError> {
+    // Note: This will swallow any errors when converting to a timestamp
+    let expiration: Option<u64> = match ttl {
+        0 => Some(0),
+        _ => {
+            let now = SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .expect("Could not retrieve UNIX timestamp");
+            now.checked_add(Duration::from_secs(u64::from(ttl)))
+                .map(|expiration| expiration.as_secs())
+        },
+    };
```

Comment:
This is what the linter suggests:

    let expiration: Option<u64> = if ttl == 0 {
        Some(0)
    } else {
        let now = SystemTime::now()
            .duration_since(SystemTime::UNIX_EPOCH)
            .expect("Could not retrieve UNIX timestamp");
        now.checked_add(Duration::from_secs(u64::from(ttl)))
            .map(|expiration| expiration.as_secs())
    };